### PR TITLE
[DOC] Document jax.checkpoint policies

### PR DESCRIPTION
**Pull request by nouiz**
Advance #12417 without finishing it.

### CODE REVIEW

File: docs/gradient-checkpointing.md [hunk 1]
```diff
@@ -354,18 +354,24 @@ print_saved_residuals(loss_checkpoint2, params, x, y)
...
+  `checkpoint_name` with any of the names given)
+* `save_any_names_but_these` (save only named values, i.e. any outputs of
+  `checkpoint_name`, except for those with the names given)
+* `save_only_these_names` (save only named values, and only among the names
+  given)
+* `offload_dot_with_no_batch_dims` same as `dots_with_no_batch_dims_saveable`,
+  but offload to CPU memory instead of recomputing.
+* `save_and_offload_only_these_names` same as `save_only_these_names`, but
+  offload to CPU memory instead of recomputing.
+* `save_from_both_policies(policy_1, policy_2)` This `or` two policies.
```
Comment (mattjj) [inline docs/gradient-checkpointing.md:374]:
```suggestion
* `save_from_both_policies(policy_1, policy_2)` (like a logical `or`, so that a residual is saveable if it is saveable according to `policy_1` _or_ `policy_2`)
```

Comment (mattjj) [reply docs/gradient-checkpointing.md:374]: Actually I think we should rename `save_from_both_policies` to `save_from_either_policy` because to me the name "both" is ambiguous as to whether it means "or" or "and". But we can do that later.

**Review by mattjj — APPROVED** [review]
Thanks for doing this!
